Fix #289190 - Toolbar customization refinements & fixes

Obliquely opened the pull request:
- made dialogue show the proper names for customisable actions / item
  previously it was showing the internal tags/ids (which won't translate)

- allow for the view mode / view zoom to be customised like other items

- fixed title of dialogue box

- removed text box showing workspace name from dialogue box - it's an
  edit box showing information that can never be edited - unhelpful UI

- added name of workspace being edited to the dialogue box title

- re-ordered the available and chosen lists in the dialogue box to make
  more sense - most specific is at the right and not in the middle

- added tooltips to various items in dialogue box

- menu is greyed out / disabled when toolbars are not customisable,
  i.e. because we're in advanced or basic mode

- available actions now maintain STANDARD order when items are removed
  from the current toolbar

Obliquely commented:
![Before changes](https://user-images.githubusercontent.com/3323784/57780129-8c1e5380-771f-11e9-8802-d3fa8695b30a.png)

![After changes](https://user-images.githubusercontent.com/3323784/57780134-8fb1da80-771f-11e9-8cbf-2bcacadc686e.png)

Jojo-Schmitz commented on mscore/musescore.cpp:
```diff
@@ -909,77 +911,81 @@ void MuseScore::populateFileOperations()
+          _fileOperationEntries.reverse();
 
-      viewModeCombo = new QComboBox(this);
+        for (auto s : _fileOperationEntries) {
```
wrong indentation, we use 6 spaces per level, here it is 8

Jojo-Schmitz commented on mscore/musescore.cpp:
```diff
@@ -909,77 +911,81 @@ void MuseScore::populateFileOperations()
+    
+    // view-options is treated as a special case as it's not a QToolButton
+    // but, rather, made up of two QComboBox objects (one sub-classed).
+    void MuseScore::addViewOptionsDoubleWidget()
```
hee it is 4

Jojo-Schmitz commented on mscore/musescore.h:
```diff
@@ -882,6 +883,7 @@ MasterSynthesizer* synthesizerFactory();
 Driver* driverFactory(Seq*, QString driver);
 
 extern QAction* getAction(const char*);
+    extern QString getShortcutText(const char*);
```
wring indent again

Jojo-Schmitz commented on mscore/shortcut.cpp:
```diff
@@ -182,6 +182,15 @@ Shortcut Shortcut::_sc[] = {
+         QT_TRANSLATE_NOOP("action","View Options"),
+         0,
+         0,
+          Icons::viewOptions_ICON,
```
different indeting here, but one space too much ;-)

Jojo-Schmitz commented on mscore/shortcut.cpp:
```diff
@@ -3879,6 +3888,18 @@ QAction* getAction(const char* id)
+
+QString getShortcutText(const char* id)
+{
+    Shortcut* shortcut = Shortcut::getShortcut(id);
```
6 spaces here please

Jojo-Schmitz commented on mscore/toolbarEditor.ui:
```diff
@@ -138,20 +117,39 @@
+   <item row="1" column="0" rowspan="2" colspan="2">
+    <widget class="QListWidget" name="toolbarList">
+     <property name="toolTip">
+      <string>Toolbar to customise</string>
```
`customize` please, US spelling

Jojo-Schmitz commented on mscore/musescore.cpp:
```diff
@@ -916,44 +916,44 @@ void MuseScore::populateFileOperations()
-              }
+            else
+                  fileTools->addWidget(new AccessibleToolButton(fileTools, getAction(s)));
+      }
```
Wrong indent ( we use banner style, closing } indented)

Jojo-Schmitz commented on mscore/musescore.cpp:
```diff
@@ -909,77 +911,81 @@ void MuseScore::populateFileOperations()
+      bool leftToRightLayout = qApp->layoutDirection() == Qt::LayoutDirection::LeftToRight;
+       
+      if (!leftToRightLayout)
+          _fileOperationEntries.reverse();
```
6 spaces here

Jojo-Schmitz commented on mscore/musescore.cpp:
```diff
@@ -909,77 +911,81 @@ void MuseScore::populateFileOperations()
+      }
+      
+      if (!leftToRightLayout)
+         _fileOperationEntries.reverse();
```
and here

Jojo-Schmitz commented on mscore/musescore.cpp:
```diff
@@ -909,77 +911,81 @@ void MuseScore::populateFileOperations()
       connect(mag, SIGNAL(magChanged(MagIdx)), SLOT(magChanged(MagIdx)));
       fileTools->addWidget(mag);
-
+      
```
no spaces here

Jojo-Schmitz commented on mscore/musescore.cpp:
```diff
@@ -909,77 +911,81 @@ void MuseScore::populateFileOperations()
 #endif
       viewModeCombo->setFixedHeight(preferences.getInt(PREF_UI_THEME_ICONHEIGHT) + 8);  // hack
-
+      
```
nor here

Jojo-Schmitz commented on mscore/musescore.cpp:
```diff
@@ -909,77 +911,81 @@ void MuseScore::populateFileOperations()
       viewModeCombo->addItem(tr("Continuous View"), int(LayoutMode::LINE));
       viewModeCombo->addItem(tr("Single Page"), int(LayoutMode::SYSTEM));
-
+      
```
nor here

Jojo-Schmitz commented on mscore/musescore.cpp:
```diff
@@ -909,77 +911,81 @@ void MuseScore::populateFileOperations()
 
 void MuseScore::populatePlaybackControls()
-      {
+   {
```
Either no spaces (for methods/functions) or 6

Jojo-Schmitz commented on mscore/musescore.cpp:
```diff
@@ -909,77 +911,81 @@ void MuseScore::populateFileOperations()
-            if (!*s)
-                  transportTools->addSeparator();
+         if (!*s) {
+            transportTools->addSeparator();
```
indent...

Jojo-Schmitz commented on mscore/toolbarEditor.cpp:
```diff
@@ -29,10 +29,14 @@ static const char* toolbars[] = {
-            editToolbars = new ToolbarEditor(this);
-            }
+         editToolbars = new ToolbarEditor(this);
+      }
```
banner style indent. Or no braces here at all for a single statement

Jojo-Schmitz commented on mscore/toolbarEditor.cpp:
```diff
@@ -111,49 +117,63 @@ void ToolbarEditor::accepted()
 
 void ToolbarEditor::populateLists(const std::list<const char*>& all, std::list<const char*>* current)
-      {
+   {
```
no or 6 spaces...

Jojo-Schmitz commented on mscore/toolbarEditor.cpp:
```diff
@@ -111,49 +117,63 @@ void ToolbarEditor::accepted()
+   {
       actionList->clear();
+      for (auto currentId : *current) {
+         QString shortcutText = getShortcutText(currentId);
```
indent... here and below

Jojo-Schmitz commented on mscore/toolbarEditor.cpp:
```diff
@@ -234,6 +268,10 @@ void ToolbarEditor::downAction()
+   
+  
+   
+   
```
excess white space

Jojo-Schmitz commented on mscore/workspace.cpp:
```diff
@@ -169,17 +169,20 @@ void MuseScore::changeWorkspace(QAction* a)
-            Workspace::currentWorkspace->save();
-
+         Workspace::currentWorkspace->save();
+      
```
spaces for nothing

Jojo-Schmitz commented on mscore/workspace.cpp:
```diff
@@ -169,17 +169,20 @@ void MuseScore::changeWorkspace(QAction* a)
-            updateIcons();
-            preferencesChanged(true);
-            }
+         updateIcons();
```
indent...

Jojo-Schmitz commented:
Please check out coding style guide, reg. indenting.
Also please squash the commits

anatoly-os commented:
Please don't squash :) It will be useful during the review process of such non trivial changes

Obliquely commented:
@Jojo-Schmitz I very much hope I've caught all the indenting errors now. I've squashed. Thanks for the help and apologies for using up your time on those kinds of error.

Jojo-Schmitz commented on mscore/shortcut.cpp:
```diff
@@ -182,6 +182,15 @@ Shortcut Shortcut::_sc[] = {
+         STATE_NORMAL,
+         "view-options",
+         QT_TRANSLATE_NOOP("action","View Options"),
+         0,
```
Please change that `0,` into `QT_TRANSLATE_NOOP("action","View options"),`, needed for zthe shortcuts dialog (and as a tooltip) and there in sentence case

Obliquely replied:
This comment got me thinking. The zoom box and view mode box don't really belong in the shortcuts list at all. There are already shortcuts for various zoom and page mode actions. So I've slightly changed tack. (See below.)

Obliquely commented:
@Jojo-Schmitz 's comment about shortcuts (above) made me realise that it doesn't make sense to use shortcuts to represent the zoom and view mode combo boxes. There are already shortcut actions related to what the dropdown boxes do. This slightly reduces scope of changes and avoids the need for a new icon. Also added option to customise the zoom box and the view mode independently and fixed some spacing issues related to those toolbar items.

Obliquely commented on mscore/toolbarEditor.cpp:
```diff
@@ -106,52 +108,101 @@ void ToolbarEditor::accepted()
+            return new QListWidgetItem(action->icon(), itemName);
+      }
+
+      qDebug()<<"ToolbarEditor does not recognize id for toolbar item: "<<QString(id);
```
Is it OK to leave the qDebug() in or is it better to do something with ASSERT (the syntax of which I'd need to look up :)). This should never be sprung, but it's pretty benign if it is.

Jojo-Schmitz replied:
IIRC gDebug()s vanish when build in RELEASE mode

Jojo-Schmitz commented:
Now Travis CI's mtests fail, check https://travis-ci.org/musescore/MuseScore/jobs/535052660#L4293-L4321

Obliquely commented on mscore/musescore.cpp:
```diff
@@ -4140,7 +4140,10 @@ void MuseScore::changeState(ScoreState val)
 
-      getAction("edit-toolbars")->setEnabled(!Workspace::currentWorkspace->readOnly());
+      Workspace* currentWorkspace = Workspace::currentWorkspace;
+      if (currentWorkspace) {
```
When running tst_runscripts test the currentWorkspace returns nullptr and MuseScore crashes before it even gets to run any of the tests. I've put in an explicit check to avoid this. It now passes 43/44 tests, which is exactly what master is currently passing. Is currentWorkspace unexpectedly being null because the tests are running MuseScore in "headless" mode (if that's the right terminology)? Might this be an issue elsewhere?

Jojo-Schmitz commented:
rebase needed

Jojo-Schmitz commented:
Another rebase needed

RobFog commented:
Is this another one for the design department and @Tantacrul to consider?

Jojo-Schmitz commented:
Code style issues. The Travis fails seems to be unrelated to this PR

Obliquely commented:
> Code style issues. The Travis fails seems to be unrelated to this PR

Code style fixed (via uncrustify - thanks :) ). Not sure i can do anything wrt TRAVIS fail.

Jojo-Schmitz commented:
No idea what to do about Travis, https://travis-ci.org/github/musescore/MuseScore/jobs/722473721#L484-L653
Seems entirely unrelated to the changer in this PR here

vpereverzev commented:
**Automatically archived due to long inactivity. Could be re-opened later once an author will come back**
